src/sources/util/http_client.rs

```diff
@@ -157,9 +165,19 @@ pub(crate) async fn call<
             }
 
             let start = Instant::now();
-            client
-                .send(request)
-                .map_err(Error::from)
+            let timeout = std::cmp::min(inputs.target_timeout, inputs.interval);
```

Comment:
This behavior should probably be called out in the user facing configuration options.

Reply:
I'm debating whether or not this behavior is desirable. On the one hand, I see how it prevents misconfigurations that could exhaust resources (e.g., when `target_timeout >> interval`). However, from a user point of view, I would find this behavior unexpected and may even want to set the timeout to a value larger than the interval in certain circumstances. I'm curious to hear others' thoughts.

Reply:
@dsmith3197 i was debating this too before sending it. happy to not have that there and just letting users control that themselves—it's not a very difficult option to change if you do experience issues, so foot-guns are not that critical.

Reply:
Two other options:
- document the risk of exhausting resources etc on the user facing docs, can make this a `warning` (and remove this taking the min logic)
- check for the timeout > interval in parsing the config and log a warning there.

The first seems like a better choice if we anticipate users complaining about not having that control.

Reply:
> check for the timeout > interval in parsing the config and log a warning there.

i think this is a practical solution here. i'll make those changes :+1: